Feature/add evolve data migration

Jamie-Unosquare opened the pull request:
Added migrations and interface settings.
Resolved conflicts with Swagger change

Logger to be added in future

Jamie-Unosquare committed:
Added Interfaces to allow for Data migration with Evolve, along with preprocessor for DB connection

Jamie-Unosquare committed:
Added Schema and updated config

Jamie-Unosquare committed:
Correcting conflicts

BrownMark commented on v2/AdminCore.DAL/Entity Framework/EntityFrameworkContext.cs:
```diff
@@ -45,6 +45,11 @@ public EntityFrameworkContext(AdminCoreContext adminCoreContext)
     /// </summary>
     public IRepository<Employee> EmployeeRepository => _EmployeeRepository ?? (_EmployeeRepository = new EntityFrameworkRepository<Employee>(this));
 
+    public IDbConnection RetrieveConnection()
```
Don't need this anymore.

BrownMark commented on v2/AdminCore.Common/Interfaces/IOutput.cs:
```diff
@@ -0,0 +1,16 @@
+
+namespace AdminCore.Common.Interfaces
+{
+    public interface IOutput
```
Let's delete this, was necessary for a console app unit tests.

BrownMark commented on v2/AdminCore.DAL/IDatabaseContext.cs:
```diff
@@ -25,5 +26,7 @@ public interface IDatabaseContext
     /// </summary>
     void SaveChanges();
+
+    IDbConnection RetrieveConnection();
```
Please delete.

BrownMark commented on v2/AdminCore.Services/Configuration/EvolveDataMigration.cs:
```diff
@@ -0,0 +1,89 @@
+    /// The <see cref="IDbConnection"/>.
+    /// </returns>
+    public virtual IDbConnection RetrieveDatabaseConnection()
+    {
```
Let's pull the connectionstring from the common IConfiguration implementation

Jamie-Unosquare committed:
Removed RetrieveConnection from EntityFrameworkContext and IDatabaseContext
Removed IOutput which was for console app tests
Expanded EvolveMigration to have a fall back and use Constants Project
Renamed DB to AdminCore
